posthog/api/query.py

```diff
@@ -63,31 +61,31 @@ def get_throttles(self):
     def create(self, request, *args, **kwargs) -> Response:
         data = self.get_model(request.data, QueryRequest)
         client_query_id = data.client_query_id or uuid.uuid4().hex
+        execution_mode = execution_mode_from_refresh(data.refresh)
+        response_status: int = status.HTTP_200_OK
 
         self._tag_client_query_id(client_query_id)
 
-        if data.async_:
-            query_status = enqueue_process_query_task(
-                team=self.team,
-                user=cast(User, self.request.user),
-                query_json=request.data["query"],
-                query_id=client_query_id,
-                refresh_requested=data.refresh or False,
-            )
-            return Response(query_status.model_dump(), status=status.HTTP_202_ACCEPTED)
+        if data.async_:  # TODO: Legacy async, use "refresh=async" instead
```

Comment:
TODO

Reply:
Since we are under control what our frontend sends, we can remove that already, right?